apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -76,4 +76,65 @@ function wpcom_global_styles_enqueue_scripts_and_styles() {
 		filemtime( plugin_dir_path( __FILE__ ) . 'dist/wpcom-global-styles.css' )
 	);
 }
+
+/**
+ * Returns the stylesheet resulting of merging core and theme data.
+ *
+ * @return string Stylesheet.
+ */
+function wpcom_get_free_global_stylesheet() {
+	// Return cached value if it can be used and exists.
+	$can_use_cached = ( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG )
+		&& ( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG )
+		&& ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST )
+		&& ! is_admin();
+
+	$transient_name = 'wpcom_free_global_styles_' . get_stylesheet();
+	if ( $can_use_cached ) {
+		$cached = get_transient( $transient_name );
+		if ( $cached ) {
+			return $cached;
+		}
+	}
+
+	$tree = new WP_Theme_JSON_Gutenberg();
+
+	// Merge theme and core data.
+	$tree->merge( WP_Theme_JSON_Resolver_Gutenberg::get_theme_data() );
+	$tree->merge( WP_Theme_JSON_Resolver_Gutenberg::get_core_data() );
+
+	// Generate the default spacing sizes presets.
+	$tree->set_spacing_sizes();
+
+	// We only get the stylesheet for variables and styles of the theme.
+	$stylesheet = $tree->get_stylesheet( array( 'variables', 'styles' ), array( 'theme' ) );
+
+	if ( $can_use_cached ) {
+		// This cache can be long since it only applies to customers that can't load global styles.
+		// Once the customer is on a paid plan this code isn't executed and will use the default
+		// Gutenberg global styles loading mechanism instead.
+		set_transient( $transient_name, $stylesheet, DAY_IN_SECONDS );
```

Comment:
The default (free) styles might change on every new theme release, but this cache would prevent them from being applied. This is specially important if the theme has released a new version to fix a critical regression. 

I think that's why `wp_enqueue_global_styles` and `wp_get_global_stylesheet` only caches these styles for a minute, so it'd be preferable to follow the same approach.
```suggestion
		set_transient( $transient_name, $stylesheet, MINUTE_IN_SECONDS );
```